pkg/deployment/deployment.go

```diff
@@ -361,6 +361,42 @@ func (builder *Builder) WithSchedulerName(schedulerName string) *Builder {
 	return builder
 }
 
+// WithHostNetwork applies a hostnetwork state to the deployment definition.
+func (builder *Builder) WithHostNetwork(enableHostnetwork bool) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Setting hostnetwork %v to deployment %s in namespace %s",
+		enableHostnetwork, builder.Definition.Name, builder.Definition.Namespace)
+
+	builder.Definition.Spec.Template.Spec.HostNetwork = enableHostnetwork
+
+	return builder
+}
+
+// WithAffinity sets affinity on deployment definition.
+func (builder *Builder) WithAffinity(affinity *corev1.Affinity) *Builder {
```

Comment:
could you please add a test for this function?

Reply:
oh I see this is from the other PR, in that case could you please remove the WithAffinity from this PR so it's just in one of them?

Reply:
sure, one sec, I was sure it was removed already

Reply:
sorry for confusing you. it was implemented twice, since it takes too much time to merge prev. MRs and I forgot that it was already implemented.
Finally I just didn't push last code with the removed duplication